packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts

```diff
@@ -558,6 +567,16 @@ export class CloudFrontWebDistribution extends cdk.Construct {
 
     }
 
+    public validate(): string[] {
```

Comment:
Is it possible to add origins after the construct is created (might be a useful feature)? If not, then why not just validate this in the constructor?

Reply:
I don't mind validating in the constructor - I just thought that was the point of the `validate` method. 

Is the idea that validate is only used if you have post-constructor modifications? (That makes sense to me, just trying to get a feel for how you see it)

Reply:
Yap. It's a hook that allows you to perform validations just before synthesis. But the sooner you can perform a validation, the better, because the error will include the most context for the user to be able to understand what they did wrong.

Reply:
Understood. I'll drop the `validate` for the time being - I don't currently think adding a origin post creation is valuable. (In my mind - it's not.)

If someone opens a pull request later - it's easy enough to add that.